canonical/maas-ui
# refactor: machine slice on deleteNotify

## Done

- refactor: machine slice on deleteNotify

### Note
Review https://github.com/canonical/maas-ui/pull/5012 first

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to machine list and set grouping to a value other than "No grouping"
- Open another instance of machine list in a second browser window
- Delete a machine in the second window
- Ensure that the machine list has been updated correctly (including group counts) in the first window

## Fixes

Fixes: https://warthogs.atlassian.net/browse/MAASENG-1858

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

It could be helpful to provide some screenshots to aid in QAing the change.
